addons/docs/src/frameworks/vue/sourceDecorator.ts

```diff
@@ -0,0 +1,198 @@
+  const attrString = [
+    ...(vnode.data?.slot ? ([['slot', vnode.data.slot]] as [string, any][]) : []),
+    ...(vnode.componentOptions?.propsData ? Object.entries(vnode.componentOptions.propsData) : []),
+    ...(vnode.data?.attrs ? Object.entries(vnode.data.attrs) : []),
+  ]
+    .filter(([name], index, list) => list.findIndex((item) => item[0] === name) === index)
+    .map(([name, value]) => stringifyAttr(name, value))
+    .filter(Boolean)
+    .join(' ');
+
+  if (!vnode.componentOptions) {
+    // Non-component elements (div, span, etc...)
+    if (vnode.tag) {
+      if (!vnode.children) {
+        return `<${vnode.tag} ${attrString}/>`;
+      }
+
+      return `<${vnode.tag} ${attrString}>${vnode.children.map(vnodeToString).join('')}</${
+        vnode.tag
+      }>`;
+    }
+
+    // TextNode
+    if (vnode.text) {
+      if (/[<>"&]/.test(vnode.text)) {
+        return `{{\`${vnode.text.replace(/`/g, '\\`')}\`}}`;
+      }
+
+      return vnode.text;
+    }
+
+    // Unknown
+    return '';
+  }
+
+  // Probably users never see the "unknown-component". It seems that vnode.tag
+  // is always set.
+  const tag = vnode.componentOptions.tag || vnode.tag || 'unknown-component';
+
+  if (!vnode.componentOptions.children) {
+    return `<${tag} ${attrString}/>`;
+  }
+
+  return `<${tag} ${attrString}>${vnode.componentOptions.children
+    .map(vnodeToString)
+    .join('')}</${tag}>`;
+}
+
+function stringifyAttr(attrName: string, value?: any): string | null {
+  if (typeof value === 'undefined') {
+    return null;
+  }
+
+  if (value === true) {
+    return attrName;
+  }
+
+  if (typeof value === 'string') {
+    return `${attrName}=${quote(value)}`;
+  }
+
+  // TODO: Better serialization (unquoted object key, Symbol/Classes, etc...)
+  //       Seems like Prettier don't format JSON-look object (= when keys are quoted)
+  return `:${attrName}=${quote(JSON.stringify(value))}`;
+}
+
+function quote(value: string) {
+  return value.includes(`"`) && !value.includes(`'`)
+    ? `'${value}'`
+    : `"${value.replace(/"/g, '&quot;')}"`;
+}
+
+/**
+ * Skip decorators and grab a story component itself.
+ * https://github.com/pocka/storybook-addon-vue-info/pull/113
+ */
+function getStoryComponent(w: any) {
+  let matched = w;
+
+  while (
+    matched &&
+    matched.options &&
+    matched.options.components &&
+    matched.options.components.story &&
+    matched.options.components.story.options &&
+    matched.options.components.story.options.STORYBOOK_WRAPS
+  ) {
+    matched = matched.options.components.story.options.STORYBOOK_WRAPS;
+  }
+  return matched;
+}
+
+interface VueInternal {
+  // We need to access this private property, in order to grab the vnode of the
+  // component instead of the "vnode of the parent of the component".
+  // Probably it's safe to rely on this because vm.$vnode is a reference for this.
+  // https://github.com/vuejs/vue/issues/6070#issuecomment-314389883
+  _vnode: Vue.VNode;
```

Comment:
I think it's safe to use this "private" property. Without this, we can't access the VNode for stories having no children such like this:

```js
// Instance for the story has zero-length `$children` and `$vnode` describes itself, not content
export const myStory = () => ({
  template: `
    <button>Button</button>
  `
})
```